### PR TITLE
Advance configurations for CDC using logical replication

### PR DESCRIPTION
**Pull request by siddharth2411**

### CODE REVIEW

File: docs/content/preview/reference/configuration/yb-tserver.md [hunk 1]
```diff
@@ -1288,6 +1288,38 @@ Number of tables to be added to the stream ID per run of the background thread w
...
+
+##### --cdcsdk_publication_list_refresh_interval_secs
+
+Interval in seconds at which the table list in the publication will be refreshed.
+
+Default: `3600`
+
+##### --cdcsdk_max_consistent_records
+
+Controls the maximum number of records sent from Virtual WAL (VWAL) to Walsender in consistent order.
```
Comment (vaibhav-yb) [inline docs/content/preview/reference/configuration/yb-tserver.md:1313]: I think `Walsender` should be referred to as `walsender` - no uppercasing of first letter.

File: docs/content/preview/reference/configuration/yb-tserver.md [hunk 2]
```diff
@@ -1288,6 +1288,38 @@ Number of tables to be added to the stream ID per run of the background thread w
...
+
+Default: `500`
+
+##### --cdcsdk_vwal_getchanges_resp_max_size_bytes
+
+Max size (in bytes) of changes sent from CDC Service to Virtual WAL (VWAL) for a particular tablet.
```
Comment (vaibhav-yb) [inline docs/content/preview/reference/configuration/yb-tserver.md:1319]: Do we have any other page where we have described what a VirtualWAL is?

**Review by vaibhav-yb — APPROVED** [review]

File: docs/content/preview/explore/change-data-capture/using-logical-replication/advanced-configuration.md [hunk 3]
```diff
@@ -1,12 +1,32 @@
...
+- [cdcsdk_enable_dynamic_table_support](../../../../reference/configuration/yb-tserver/#cdcsdk-enable-dynamic-table-support)
+- [cdcsdk_publication_list_refresh_interval_secs](../../../../reference/configuration/yb-tserver/#cdcsdk-publication-list-refresh-interval-secs)
+- [cdcsdk_max_consistent_records](../../../../reference/configuration/yb-tserver/#cdcsdk-max-consistent-records)
+- [cdcsdk_wal_getchanges_resp_max_size_bytes](../../../../reference/configuration/yb-tserver/#cdcsdk-wal-getchanges-resp-max-size-bytes)
+
+## Retention of Resources
+
+CDC retains resources, such as WAL segments, that contain information related to the changes involved in the transactions. These resources are typically retained until the consuming client acknowledges the receipt of all the transactions contained in that resource. Retaining resources has an impact on the system. Clients are expected to consume these transactions within configurable duration limits. Resources will be released if the duration exceeds these configured limits. The following are the configuration Gflags that control the duration limits for which resources are retained:
```
Comment (vaibhav-yb) [inline docs/content/preview/explore/change-data-capture/using-logical-replication/advanced-configuration.md:27]:
```suggestion
CDC retains resources such as WAL segments that contain information related to the changes involved in the transactions. These resources are typically retained until the consuming client acknowledges the receipt of all the transactions contained in that resource. Retaining resources has an impact on the system. Clients are expected to consume these transactions within configurable duration limits. Resources will be released if the duration exceeds these configured limits. The following are the configuration Gflags that control the duration limits for which resources are retained:
```

File: docs/content/preview/reference/configuration/yb-tserver.md [hunk 4]
```diff
@@ -1296,6 +1296,38 @@ Number of tables to be added to the stream ID per run of the background thread w
 
 Default: `2`
 
+Following set of flags are only relevant for CDC using logical replication. To learn about CDC using logical replication, see [Change data capture (CDC) using logical replication](../../../architecture/docdb-replication/cdc-logical-replication).
+
+##### --ysql_yb_default_replica_identity
+
+The default replica identity to be assigned to user defined tables at the time of creation. The flag is case sensitive and can take four possible values, `FULL`, `DEFAULT`,`'NOTHING` and `CHANGE`. If any value other than these is assigned to the flag, the replica identity `CHANGE` will be used as default at the time of table creation.
```
Comment (vaibhav-yb) [inline docs/content/preview/reference/configuration/yb-tserver.md:1303]: > If any value other than these is assigned to the flag, the replica identity `CHANGE` will be used as default at the time of table creation.

Can you please point me to the code for this logic?

**Review by vaibhav-yb — APPROVED** [review]

File: docs/content/preview/reference/configuration/yb-tserver.md (hunk 2 above)
Comment (siddharth2411) [reply docs/content/preview/reference/configuration/yb-tserver.md:1327]: We don't have a dedicated subsection just for VWAL. But in the architecture page, we do talk Virtual WAL and its responsibilities. Should we link the architecture section itself?

File: docs/content/preview/reference/configuration/yb-tserver.md (hunk 4 above)
Comment (siddharth2411) [reply docs/content/preview/reference/configuration/yb-tserver.md:1303]: Checked the code and currently it seems to return an error if the user does not pass any of the above four values. Confirmed the behavior with Sumukh as well. Therefore, removing this line. 

Note: The description of flag in code also has to be changed (will be done by sumukh in a separate diff). 

Good catch!

**Pull request merged**